cpp/src/arrow/ipc/adapter.cc

```diff
@@ -41,6 +42,7 @@ namespace flatbuf = apache::arrow::flatbuf;
 namespace ipc {
 
 static bool IsPrimitive(const DataType* type) {
+  DCHECK(type != nullptr);
```

Comment:
Do you prefer this or `DCHECK(type)`?  For checks like this I think I prefer this without explicit comparison, but its not a strong preference.